hello-world/tutorial/index.md

```diff
@@ -0,0 +1,471 @@
+
+```go
+send-ibcPost <portID> <channelD> <title> <content>
+```
+
+## Source code modification
+
+After creating our types and transactions, we have to manually insert the logic accordingly to what happens in our datatables.
+Let's get into the code and make the modifications to save the data according to what we have defined above.
+
+### Adding post creator to the packet
+
+Let's start with the proto file, which defines the structure of our IBC packet.
+We want the receiving blockchain to identify the creator of the post. We add the creator field inside the packet (we didn't specify this field directly in the command because it would automatically become a parameter in `SendIbcPost` CLI command)
+
+```go
+// planet/proto/blog/packet.proto
+// this line is used by starport scaffolding # ibc/packet/proto/message
+// IbcPostPacketData defines a struct for the packet payload
+message IbcPostPacketData {
+	string title = 1;
+    string content = 2;
+    string creator = 3; // < ---
+}
+```
+
+We want to make sure the receiving chain has content on the creator of a blog post. We add this value to our IBC `packet`.
+`SendIbcPost` message automatically has the content of the `sender` of the message, this value is already used and verified as the signer of the message. We can therefore add the `msg.Sender` as creator to the new packet before sending it over IBC
+
+```go
+// planet/x/blog/keeper/msg_server_ibcPost.go
+// Construct the packet
+	var packet types.IbcPostPacketData
+
+	packet.Title = msg.Title
+	packet.Content = msg.Content
+	packet.Creator = msg.Sender // < ---
+	
+	// Transmit the packet
+	err := k.TransmitIbcPostPacket(
+		ctx,
+		packet,
+		msg.Port,
+		msg.ChannelID,
+		clienttypes.ZeroHeight(),
+		msg.TimeoutTimestamp,
+	)
+```
+
+## Receiving the post
+
+We are at the heart of our transaction logic, located inside `planet/x/blog/keeper/ibcPost.go`
+
+This file contain three methods that will help us learn the core methods about IBC packets.
+
+- `TransmitIbcPostPacket` called manually to send the packet over IBC, it can also define the logic before the packet is sent over IBC to another blockchain app
+- `OnRecvIbcPostPacket` hook (automatically called) when a packet is received on the chain, we define here the logic of packet reception
+- `OnAcknowledgementIbcPostPacket` hook when a sent packet is acknowledged on the source chain, we define here logics when we know the packet has been received
+- `OnTimeoutIbcPostPacket` hook when a sent packet is timed out, we define here logics when we know the packet will never be received on the target chain
+
+What we need to add into the source code is the logic inside those functions to add to the datatables accordingly.
+
+On reception of the message, we create a new post with the title and the content on the receiving chain.
+
```

Comment:
```suggestion
You must modify the source code to add the logic inside those functions so that the data tables are modified accordingly.

On reception of the message, we create a new post with the title and the content on the receiving chain.

```